tls/_common/prf.py

```diff
@@ -0,0 +1,69 @@
+# This file is dual licensed under the terms of the Apache License, Version
+# 2.0, and the BSD License. See the LICENSE file in the root of this repository
+# for complete details.
+
+from __future__ import absolute_import, division, print_function
+
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives.hmac import HMAC
+
+
+def _p_hash(hash_algorithm, secret, seed, output_length):
+    """
+    A seed expansion function that uses a single hash function to expand a
+    secret and seed into the number of bytes specified by output_length.
+    """
+    result = b""
+    i = 1
+    while len(result) < output_length:
+        h = HMAC(secret, hash_algorithm, default_backend())
+        h.update(_a(secret, hash_algorithm, i, seed))
+        h.update(seed)
+        result += h.finalize()
```

Comment:
So to replicate the behaviour of this loop might be hard without using `list.append` which I'm not convinced would be less awful than just concatenating bytes objects together. Instead I think we should do something like:

```py
result = bytesarray()
i = 1
while len(result) < output_length:
    h = HMAC(secret, hash_algorithm, default_backend())
    h.update(_a(secret, hash_algorithm, i, seed))
    h.update(seed)
    result.extend(h.finalize())
    i += 1
return bytes(result[:output_length])
```